apps-api/backoffice/src/modules/admin/election/repository.ts

```diff
@@ -447,37 +439,132 @@ export class AdminElectionRepository {
     )
   }
 
-  async upsertElectionOnsiteResult(result: { candidateId: string; votes: number }[]) {
+  async countElectionEligibleVoters(electionId: string, type: EligibleVoterType) {
+    return fromRepositoryPromise(
+      this.prismaService.electionEligibleVoter.count({
+        where: {
+          electionId,
+          type,
+        },
+      })
+    )
+  }
+
+  private async upsertElectionResult(
+    tx: Prisma.TransactionClient,
+    electionId: string,
+    type: ElectionResultType,
+    result: { candidateId: string; votes: number }[]
+  ) {
+    await tx.electionResult.updateMany({
+      where: {
+        candidate: {
+          electionId,
+        },
+        type,
+      },
+      data: {
+        count: 0,
+      },
+    })
+
+    for (let i = 0; i < result.length; i += 5) {
+      const endIdx = Math.min(i + 5, result.length)
+      const batch = result.slice(i, endIdx)
+
+      await Promise.all(
+        batch.map(({ candidateId, votes }) =>
+          tx.electionResult.upsert({
+            where: {
+              candidateId_type: {
+                candidateId,
+                type,
+              },
+            },
+            create: {
+              candidateId,
+              type,
+              count: votes,
+            },
+            update: {
+              count: votes,
+            },
+          })
+        )
+      )
+    }
```

Comment:
Can we use https://tanstack.com/pacer/latest/docs/guides/batching#batching-concept ?